src/GoogleAdWordsEventForwarder.js

```diff
@@ -70,51 +71,151 @@
+            var adWordEvent = {};
+            adWordEvent.google_conversion_value = 0;
+            adWordEvent.google_conversion_language = 'en';
+            adWordEvent.google_conversion_format = '3';
+            adWordEvent.google_conversion_color = 'ffffff';
+            adWordEvent.google_remarketing_only = forwarderSettings.remarketingOnly == 'True';
+            adWordEvent.google_conversion_id = forwarderSettings.conversionId;
+            return adWordEvent;
+        }
+
+        function generateAdwordsEvent(mPEvent, conversionLabel, isPageEvent) {
+            var adWordEvent = getBaseAdWordEvent();
+            adWordEvent.google_conversion_label = conversionLabel;
+            adWordEvent.google_custom_params = getCustomProps(mPEvent, isPageEvent);
 
-                var adWordEvent = getBaseAdWordEvent();
-                adWordEvent.google_conversion_label = conversionLabel;
+            return adWordEvent;
+        }
+
+        function generateAdwordsCommerceEvent(mPEvent, conversionLabel, isPageEvent) {
+            var adWordEvent = getBaseAdWordEvent();
+            adWordEvent.google_conversion_label = conversionLabel;
+
+            if (mPEvent.ProductAction.ProductActionType === mParticle.ProductActionType.Purchase
+                && mPEvent.ProductAction.TransactionId) {
+                adWordEvent.google_conversion_order_id = mPEvent.ProductAction.TransactionId;
+            }
+
+            if (mPEvent.CurrencyCode) {
+                adWordEvent.google_conversion_currency = mPEvent.CurrencyCode;
+            }
 
+            if (mPEvent.ProductAction.TotalAmount) {
+                adWordEvent.google_conversion_value = mPEvent.ProductAction.TotalAmount;
+            }
 
+            adWordEvent.google_custom_params = getCustomProps(mPEvent, isPageEvent);
+            return adWordEvent;
+        }
+
+        // gtag Events
+        function generateGtagEvent(mPEvent, conversionLabel, isPageEvent) {
+            var conversionPayload = {
+                'send-to': gtagSiteId + '/' + conversionLabel
+            };
+
+            var customProps = getCustomProps(mPEvent, isPageEvent);
+
+            var payload = Object.assign({}, conversionPayload, customProps);
+
+            return payload;
+        }
+
+        function generateGtagCommerceEvent(mPEvent, conversionLabel, isPageEvent) {
+            var conversionPayload = {
+                'send-to': gtagSiteId + '/' + conversionLabel
+            };
+
+            var customProps = getCustomProps(mPEvent, isPageEvent);
+
+            if (mPEvent.ProductAction.ProductActionType === mParticle.ProductActionType.Purchase
+                && mPEvent.ProductAction.TransactionId) {
                 if (event.ProductAction.ProductActionType === mParticle.ProductActionType.Purchase
                     && event.ProductAction.TransactionId) {
                     adWordEvent.google_conversion_order_id = event.ProductAction.TransactionId;
                 }
```

Comment:
this codeblock here needs to change instances of `event` --> `mPEvent`.  this is why tests are failing